### PR TITLE
Use conv2dByMatMul when input and filter have the same width/height

### PR DESCRIPTION
**Pull request by qjia7**
This PR 1) moves the common part of Conv2D/FusedConv2D to conv2DImpl to reuse the code; 2) adds another situation that input width and height are equal to filter width and height, and the pad type is valid, we can also go to the `conv2dByMatMul` path;
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6028)

### CODE REVIEW

**Comment by qjia7** [comment]
With this change, a FusedConv2d with input0: 4D[1,2,2,32], input1: 4D[2,2,32,1] will call the matmul kernel(0.03ms) instead of conv2d kernel(0.12ms). 

@gyagp @xhcao @axinging @haoyunfeix Please take a look, thanks.

File: tfjs-backend-webgpu/src/kernels/Conv2D_impl.ts [hunk 1]
```diff
@@ -183,3 +207,86 @@ export function conv2dWithIm2Col({
...
+      convInfo,
+      backend,
+      bias,
+      activation,
+      preluActivationWeights,
+      leakyreluAlpha
+    });
+  }
+
+  if (env().getBool('WEBGPU_CONV_SEPARATE_IM2COL_SHADER') && x.shape[0] === 1) {
```
Comment (xhcao) [inline tfjs-backend-webgpu/src/kernels/Conv2D_impl.ts:245]: In original code, this flag is only used by Conv2D. Could fusedConv2D also use conv2dWithIm2Col here?

Comment (qjia7) [reply tfjs-backend-webgpu/src/kernels/Conv2D_impl.ts:245]: Yes, it also works for `fusedConv2d`.

**Comment by xhcao** [comment]
LGTM, thank you.

**Review by gyagp — APPROVED** [review]
LGTM

**Comment by qjia7** [comment]
cc @lina128 @pyu10055

**Pull request merged**